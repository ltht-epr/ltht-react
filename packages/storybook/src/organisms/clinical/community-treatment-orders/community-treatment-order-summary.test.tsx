import React from 'react'
import { mount } from 'enzyme'
import ReactDOM from 'react-dom'

import CommunityTreatmentOrderSummary from '@ltht-react/community-treatment-order-summary'
import orders from './community-treatment-order.fixtures'

const Sut: React.FC = () => {
  return (
    <>
      {orders.map(order => (
        <div key={order.id}>
          <CommunityTreatmentOrderSummary communityTreatmentOrder={order} />
        </div>
      ))}
    </>
  )
}

describe('CommunityTreatmentOrderSummary', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Sut />, div)
  })
  it('matches snapshot', () => {
    expect(mount(<Sut />)).toMatchSnapshot('wrapper mount')
  })
})
