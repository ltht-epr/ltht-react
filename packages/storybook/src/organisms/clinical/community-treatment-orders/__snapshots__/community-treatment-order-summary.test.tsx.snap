// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommunityTreatmentOrderSummary matches snapshot: wrapper mount 1`] = `
.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-0 {
  color: #231f20;
  text-align: left;
}

.emotion-1 {
  color: #768692;
  text-align: left;
  font-size: smaller;
  padding-top: 0.25rem;
}

.emotion-5 {
  text-align: right;
}

.emotion-3 {
  color: #231f20;
}

.emotion-4 {
  color: #768692;
  text-align: right;
  font-size: smaller;
  padding-top: 0.25rem;
}

.emotion-8 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  text-align: left;
}

.emotion-7 {
  color: #768692;
}

.emotion-10 {
  text-align: right;
}

<Sut>
  <div
    key="580ca927-34e0-e911-a2c7-005056926fe4|communityTreatmentOrder"
  >
    <CommunityTreatmentOrderSummary
      communityTreatmentOrder={
        Object {
          "consentToTreat": "Consent Not Given",
          "id": "580ca927-34e0-e911-a2c7-005056926fe4|communityTreatmentOrder",
          "legalStatus": Object {
            "coding": Array [
              Object {
                "code": "3309007",
                "display": "Approved",
              },
            ],
            "text": "Approved",
          },
          "metadata": Object {
            "dataSources": Array [
              Object {
                "display": "Mock",
              },
            ],
            "isRedacted": false,
            "requestedWhen": "",
          },
          "period": Object {
            "end": Object {
              "kind": "DATE",
              "value": "2014-02-02T00:00:00+00:00",
            },
            "start": Object {
              "kind": "DATE",
              "value": "2013-01-01T00:00:00+00:00",
            },
          },
          "restrictions": "No restrictions",
        }
      }
    >
      <Styled(div)>
        <div
          className="emotion-6"
        >
          <Styled(div)>
            <div
              className="emotion-2"
            >
              <CommunityTreatmentOrderConsent
                communityTreatmentOrder={
                  Object {
                    "consentToTreat": "Consent Not Given",
                    "id": "580ca927-34e0-e911-a2c7-005056926fe4|communityTreatmentOrder",
                    "legalStatus": Object {
                      "coding": Array [
                        Object {
                          "code": "3309007",
                          "display": "Approved",
                        },
                      ],
                      "text": "Approved",
                    },
                    "metadata": Object {
                      "dataSources": Array [
                        Object {
                          "display": "Mock",
                        },
                      ],
                      "isRedacted": false,
                      "requestedWhen": "",
                    },
                    "period": Object {
                      "end": Object {
                        "kind": "DATE",
                        "value": "2014-02-02T00:00:00+00:00",
                      },
                      "start": Object {
                        "kind": "DATE",
                        "value": "2013-01-01T00:00:00+00:00",
                      },
                    },
                    "restrictions": "No restrictions",
                  }
                }
              >
                <div
                  className="emotion-0"
                >
                  Approved
                </div>
              </CommunityTreatmentOrderConsent>
              <CommunityTreatmentOrderSummaryRestrictions
                communityTreatmentOrder={
                  Object {
                    "consentToTreat": "Consent Not Given",
                    "id": "580ca927-34e0-e911-a2c7-005056926fe4|communityTreatmentOrder",
                    "legalStatus": Object {
                      "coding": Array [
                        Object {
                          "code": "3309007",
                          "display": "Approved",
                        },
                      ],
                      "text": "Approved",
                    },
                    "metadata": Object {
                      "dataSources": Array [
                        Object {
                          "display": "Mock",
                        },
                      ],
                      "isRedacted": false,
                      "requestedWhen": "",
                    },
                    "period": Object {
                      "end": Object {
                        "kind": "DATE",
                        "value": "2014-02-02T00:00:00+00:00",
                      },
                      "start": Object {
                        "kind": "DATE",
                        "value": "2013-01-01T00:00:00+00:00",
                      },
                    },
                    "restrictions": "No restrictions",
                  }
                }
              >
                <div
                  className="emotion-1"
                >
                  No restrictions
                </div>
              </CommunityTreatmentOrderSummaryRestrictions>
            </div>
          </Styled(div)>
          <Styled(div)>
            <div
              className="emotion-5"
            >
              <PeriodSummary
                period={
                  Object {
                    "end": Object {
                      "kind": "DATE",
                      "value": "2014-02-02T00:00:00+00:00",
                    },
                    "start": Object {
                      "kind": "DATE",
                      "value": "2013-01-01T00:00:00+00:00",
                    },
                  }
                }
              >
                <div
                  className="emotion-3"
                >
                  Jan-01,-2013 to Feb-02,-2014
                </div>
              </PeriodSummary>
              <CommunityTreatmentOrderSummaryConsent
                communityTreatmentOrder={
                  Object {
                    "consentToTreat": "Consent Not Given",
                    "id": "580ca927-34e0-e911-a2c7-005056926fe4|communityTreatmentOrder",
                    "legalStatus": Object {
                      "coding": Array [
                        Object {
                          "code": "3309007",
                          "display": "Approved",
                        },
                      ],
                      "text": "Approved",
                    },
                    "metadata": Object {
                      "dataSources": Array [
                        Object {
                          "display": "Mock",
                        },
                      ],
                      "isRedacted": false,
                      "requestedWhen": "",
                    },
                    "period": Object {
                      "end": Object {
                        "kind": "DATE",
                        "value": "2014-02-02T00:00:00+00:00",
                      },
                      "start": Object {
                        "kind": "DATE",
                        "value": "2013-01-01T00:00:00+00:00",
                      },
                    },
                    "restrictions": "No restrictions",
                  }
                }
              >
                <div
                  className="emotion-4"
                >
                  Consent Not Given
                </div>
              </CommunityTreatmentOrderSummaryConsent>
            </div>
          </Styled(div)>
        </div>
      </Styled(div)>
    </CommunityTreatmentOrderSummary>
  </div>
  <div
    key="690ca927-68e0-c511-b2c7-898056926ze4|communityTreatmentOrder"
  >
    <CommunityTreatmentOrderSummary
      communityTreatmentOrder={
        Object {
          "consentToTreat": "Consent Given",
          "id": "690ca927-68e0-c511-b2c7-898056926ze4|communityTreatmentOrder",
          "legalStatus": Object {
            "coding": Array [
              Object {
                "code": "779007",
                "display": "Pending",
              },
            ],
            "text": "Pending",
          },
          "metadata": Object {
            "dataSources": Array [
              Object {
                "display": "Mock",
              },
            ],
            "isRedacted": true,
            "requestedWhen": "",
          },
          "period": Object {
            "start": Object {
              "kind": "DATE",
              "value": "2020-01-02T00:00:00+00:00",
            },
          },
          "restrictions": "No Restrictions",
        }
      }
    >
      <Styled(div)>
        <div
          className="emotion-6"
        >
          <CommunityTreatmentOrderRedacted
            communityTreatmentOrder={
              Object {
                "consentToTreat": "Consent Given",
                "id": "690ca927-68e0-c511-b2c7-898056926ze4|communityTreatmentOrder",
                "legalStatus": Object {
                  "coding": Array [
                    Object {
                      "code": "779007",
                      "display": "Pending",
                    },
                  ],
                  "text": "Pending",
                },
                "metadata": Object {
                  "dataSources": Array [
                    Object {
                      "display": "Mock",
                    },
                  ],
                  "isRedacted": true,
                  "requestedWhen": "",
                },
                "period": Object {
                  "start": Object {
                    "kind": "DATE",
                    "value": "2020-01-02T00:00:00+00:00",
                  },
                },
                "restrictions": "No Restrictions",
              }
            }
          >
            <div
              className="emotion-8"
            >
              <RedactedDescription>
                <div
                  className="emotion-7"
                >
                  Insufficient privileges
                </div>
              </RedactedDescription>
            </div>
            <div
              className="emotion-10"
            >
              <PeriodSummary
                period={
                  Object {
                    "start": Object {
                      "kind": "DATE",
                      "value": "2020-01-02T00:00:00+00:00",
                    },
                  }
                }
              >
                <div
                  className="emotion-3"
                >
                  Jan-02,-2020 to Present
                </div>
              </PeriodSummary>
            </div>
          </CommunityTreatmentOrderRedacted>
        </div>
      </Styled(div)>
    </CommunityTreatmentOrderSummary>
  </div>
</Sut>
`;
